ang/civicase/ContactCaseTabCaseDetails.js

```diff
@@ -0,0 +1,29 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  module.directive('civicaseContactCaseTabCaseDetails', function () {
+    return {
+      restrict: 'EA',
+      replace: true,
+      controller: ContactCaseTabCaseDetailsController,
+      templateUrl: '~/civicase/ContactCaseTabCaseDetails.html',
+      scope: {
+        item: '=selectedCase'
+      }
+    };
+  });
+
+  function ContactCaseTabCaseDetailsController ($scope) {
+    $scope.getContactRole = function (caseObj) {
+      var contact = _.find(caseObj.contacts, {
+        contact_id: getSelectedContactId()
+      });
+      return contact ? contact.role : 'No Role Associated';
+    };
+    function getSelectedContactId () {
```

Comment:
Doc missing, there should be a line break between functions. and there is a separate service for this right?